inference-engine/src/transformations/src/transformations/mish_fusion.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/mish_fusion.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset4.hpp>
+#include <ngraph/rt_info.hpp>
+
+void ngraph::pass::MishFusion::mish_fusion() {
+    auto input0 = std::make_shared<pattern::op::Label>(element::f64, Shape{1, 1, 1, 1});
+    auto exp = std::make_shared<ngraph::opset4::Exp>(input0);
+    auto input_const = op::Constant::create(element::f64, Shape{1}, {-1});
+    auto add = std::make_shared<ngraph::opset4::Add>(exp, input_const);
+    auto log = std::make_shared<ngraph::opset4::Log>(add);
+    auto tanh = std::make_shared<ngraph::opset4::Tanh>(log);
+    auto mul = std::make_shared<ngraph::opset4::Multiply>(input0, tanh);
+
+    ngraph::graph_rewrite_callback callback = [](pattern::Matcher& m) {
+
+        auto mul = std::dynamic_pointer_cast<ngraph::opset4::Multiply> (m.get_match_root());
```

Comment:
@GlebKazantaev , do we really need all these checks that the operations are of the specific type? Should pattern matcher take care of that?

Reply:
See my answer below.